From 5dadef08f0504988807bc59eed9e37f89d4f84cb Mon Sep 17 00:00:00 2001
From: Werner Koch <wk@gnupg.org>
Date: Thu, 31 Jul 2025 11:37:08 +0200
Subject: [PATCH 2/2] gpgtar: Do not close stderr when calling gpg.

* tools/gpgtar-extract.c (gpgtar_extract): Do not close stderr.
* tools/gpgtar-list.c (gpgtar_list): Ditto.
--

This is another regression which made it impossible to use use
--status-fd=2 and also swallowed all other stderr output from gpg.

This is somewhat related
GnuPG-bug-id: 7757
---
 NEWS                   | 1 +
 tools/gpgtar-extract.c | 3 ++-
 tools/gpgtar-list.c    | 3 ++-
 3 files changed, 5 insertions(+), 2 deletions(-)

--- a/NEWS
+++ b/NEWS
@@ -8,6 +8,7 @@ Noteworthy changes in version 2.5.11 (20
 
   Release-info: https://dev.gnupg.org/T7743
 
+  * gpgtar: Fix regression in end-of-archive detection.  [T7757]
 
 Noteworthy changes in version 2.5.10 (2025-07-25)
 -------------------------------------------------
--- a/tools/gpgtar-extract.c
+++ b/tools/gpgtar-extract.c
@@ -492,7 +492,8 @@ gpgtar_extract (const char *filename, in
 #endif
       err = gpgrt_process_spawn (opt.gpg_program, argv,
                                  ((filename ? 0 : GPGRT_PROCESS_STDIN_KEEP)
-                                  | GPGRT_PROCESS_STDOUT_PIPE), act, &proc);
+                                  | GPGRT_PROCESS_STDOUT_PIPE
+                                  | GPGRT_PROCESS_STDERR_KEEP), act, &proc);
       gpgrt_spawn_actions_release (act);
       xfree (argv);
       if (err)
--- a/tools/gpgtar-list.c
+++ b/tools/gpgtar-list.c
@@ -527,7 +527,8 @@ gpgtar_list (const char *filename, int d
 #endif
       err = gpgrt_process_spawn (opt.gpg_program, argv,
                                  ((filename ? 0 : GPGRT_PROCESS_STDIN_KEEP)
-                                  | GPGRT_PROCESS_STDOUT_PIPE), act, &proc);
+                                  | GPGRT_PROCESS_STDOUT_PIPE
+                                  | GPGRT_PROCESS_STDERR_KEEP), act, &proc);
       gpgrt_spawn_actions_release (act);
       xfree (argv);
       if (err)
